### DISCUSSION TITLE
Is there a reason why LXQtTranslateDesktopYaml was made in Perl instead CMake?

### DISCUSSION THREAD

**Discussion by redtide** [lxqt/lxqt#2541] [Q&A] [answered]
Asking because if possible I would like to make it using CMake to be used also on Windows for my tools fork, and if accepted eventually propose a PR here.

**Comment by palinek** [comment, marked as answer]
I was involved in creation of it and the simple answer is, that I wasn't so brave to do the file reading/manipulation/writing in CMake and it was "easy" to achieve it in some well-spread scripting tool.

Refs:
https://github.com/lxqt/lxqt/issues/1044
https://github.com/lxqt/lxqt-build-tools/pull/59

**Reply by redtide** [reply]
Thank you for the quick answer!  Looking at it seemed mostly regexes, variables and file manipulation, which CMake should be able to do, but I wasn't sure if it was because some CMake limitation that I didn't know about.
